relay-profiling/src/lib.rs

```diff
@@ -126,26 +128,48 @@ struct MinimalProfile {
 }
 
 fn minimal_profile_from_json(payload: &[u8]) -> Result<MinimalProfile, ProfileError> {
-    serde_json::from_slice(payload).map_err(ProfileError::InvalidJson)
+    let d = &mut Deserializer::from_slice(payload);
+    serde_path_to_error::deserialize(d).map_err(ProfileError::InvalidJson)
 }
 
 pub fn parse_metadata(payload: &[u8]) -> Result<(), ProfileError> {
     let profile = match minimal_profile_from_json(payload) {
         Ok(profile) => profile,
-        Err(err) => return Err(err),
+        Err(err) => {
+            relay_log::warn!(error = &err as &dyn Error, "invalid minimal profile");
+            return Err(err);
+        }
     };
     match profile.version {
         sample::Version::V1 => {
-            let _: sample::ProfileMetadata = match serde_json::from_slice(payload) {
+            let d = &mut Deserializer::from_slice(payload);
+            let _: sample::ProfileMetadata = match serde_path_to_error::deserialize(d) {
                 Ok(profile) => profile,
-                Err(err) => return Err(ProfileError::InvalidJson(err)),
+                Err(err) => {
+                    relay_log::warn!(
+                        error = &err as &dyn Error,
+                        "invalid sample (v{:?}) profile for {}. invalid key: {}",
+                        profile.version,
+                        profile.platform,
+                        err.path(),
```

Comment:
Have you tried if you actually need to add `err.path()` explicitly here? the `error` passed as first argument should already give you all the error information in the logs.